record-editor/src/app/core/services/record-api.service.ts

```diff
@@ -71,6 +74,41 @@ export class RecordApiService extends CommonApiService {
       .catch((error) => Observable.throw(new ApiError(error)));
   }
 
+  async closeCuratedRecordTickets(
+    record: object,
+    pidType: string,
+    pidValue: string | number
+  ) {
+    if (record['curated']) {
+      try {
+        const tickets = await this.apiService.fetchRecordTickets(
+          pidType,
+          pidValue
+        );
+        const filteredTickets = tickets.filter((ticket) => {
+          return (ticket as SnowTicket).u_functional_category !== undefined
+            ? ticket['u_functional_category'] === 'arXiv curation' ||
+                ticket['u_functional_category'] === 'Publisher curation'
+            : ticket['queue'] === 'arXiv curation' ||
+                ticket['queue'] === 'Publisher curation';
+        });
+        for (const ticket of filteredTickets) {
+          try {
+            await this.apiService.resolveTicket(pidValue, ticket['sys_id']);
+          } catch (error) {
+            console.error('Could not close ticket', ticket['sys_id']);
+          }
+        }
+      } catch (error) {
+        if (error.status === 403) {
+          console.error('Logged in user cannot access tickets');
+        } else {
+          console.error('Could not load tickets');
+        }
+      }
+    }
+  }
```

Comment:
It's better to use use guard clauses, to make the code more readable :)
```suggestion
    if (!record['curated']) {
       return ...
    }
    [....]
```